### PR TITLE
Add optional auth to job logs

### PR DESCRIPTION
**Pull request by raskchanky**
tenor-10369144

Closes https://github.com/habitat-sh/habitat/issues/2867
Signed-off-by: Josh Black <raskchanky@gmail.com>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by reset** [comment]
@thesentinels approve

tenor-149109605

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

File: components/builder-api/src/server/handlers.rs [hunk 1]
```diff
@@ -234,27 +235,56 @@ pub fn job_log(req: &mut Request) -> IronResult<Response> {
...
+            let mut opg = OriginPackageGet::new();
+            opg.set_ident(ident);
+
+            if session_id.is_some() {
+                opg.set_account_id(session_id.unwrap());
+            }
+
+            match route_message::<OriginPackageGet, OriginPackage>(req, &opg) {
+                Ok(_) => {
+                    // We made it this far, which means either the package is public or it's
```
Comment (reset) [inline components/builder-api/src/server/handlers.rs]: I really like how this shook out. Really nice that getting past the PERMISSION NetError means you've got permissions ;)

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/281577431).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/281577431).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**